### DISCUSSION TITLE
Failed to allocate OT message on calling otUdpNewMessage at 100ms rate for longer duration

### DISCUSSION THREAD

**Discussion by nikhilkomalan** [openthread/openthread#8759] [Q&A] [answered]
Hello,
I am currently working with one of mqttsn examples based on nRF5 SDK for Thread. I am currently trying to publish data at interval of 100ms continuously. It works fine for certain minutes but after some time, **otUdpNewMessage** gives error of "Failed to allocate OT message".

So i would like to know is there any way of increasing the message buffers used by **otUdpNewMessage** ? so that i can try and avoid failed to allocate OT message error. 
```
        p_msg = otUdpNewMessage(m_p_instance, NULL);
        if (p_msg == NULL)
        {
            NRF_LOG_ERROR("Failed to allocate OT message\r\n");
            err_code = NRF_ERROR_NO_MEM;
            break;
        }
```

**Comment by jwhui** [comment, marked as answer]
See https://github.com/openthread/openthread/blob/d28793c104ab710f692a160377a5753373ae4b69/src/core/config/misc.h#L170

Note that if the average rate of generating messages exceeds the rate at which the network can deliver, simply increasing the number of message buffers will not avoid buffer depletion. However, if you are hoping to absorb bursts or variability, then increasing number of message buffers may help.

**Reply by nikhilkomalan** [reply]
How can i measure average rate of generating messages exceeds the rate at which the network can deliver ? Currently i have only one sensor node publishing data at 100 ms rate and 2-3 other sensor node publishing data at 1 minute per sample interval. 

And do i have to increase the heap allocation somewhere or is it taken care of ?

**Reply by jwhui** [reply]
> How can i measure average rate of generating messages exceeds the rate at which the network can deliver ? Currently i have only one sensor node publishing data at 100 ms rate and 2-3 other sensor node publishing data at 1 minute per sample interval.

Thread should be able to support this. Network capacity depends a lot on the specific environment. If you are operating anywhere near capacity, you'll need to run experiments in your specific environment.

> And do i have to increase the heap allocation somewhere or is it taken care of ?

Unless `OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE` is set to `1`, the message buffer pool is not dependent on the heap.
